syntax = "proto3";

package svg_proto;

message Rgb {
	uint32 red = 1;
	uint32 green = 2;
	uint32 blue = 3;
}

message Rgba {
	uint32 red = 1;
	uint32 green = 2;
	uint32 blue = 3;
	double opacity = 4;
}

message Color {
  oneof color_oneof {
    string c_string = 1;
    Rgb c_rgb = 2;
    Rgba c_rgba = 3;
  }
}